src/Sheet/FingersCrossed/Loader.php

```diff
@@ -6,27 +6,33 @@
 
 use Box\Spout\Common\Entity\Cell;
 use Box\Spout\Common\Entity\Row;
-use Box\Spout\Writer\WriterInterface;
+use Box\Spout\Writer\XLSX\Writer as XLSXWriter;
+use Box\Spout\Writer\ODS\Writer as ODSWriter;
 use Kiboko\Component\Bucket\AcceptanceResultBucket;
 use Kiboko\Component\Bucket\EmptyResultBucket;
 use Kiboko\Contract\Bucket\ResultBucketInterface;
 use Kiboko\Contract\Pipeline\FlushableInterface;
 use Kiboko\Contract\Pipeline\LoaderInterface;
 use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
 
 final class Loader implements LoaderInterface, FlushableInterface
 {
-    private ?LoggerInterface $logger = null;
+    private LoggerInterface $logger;
 
     public function __construct(
-        private WriterInterface $writer,
-        private string $sheetName
+        private XLSXWriter|ODSWriter $writer,
```

Comment:
Pourquoi est-ce que tu n'as pas conservé le `WriterInterface` ?

Reply:
Parce que quand j'utilise une interface, je ne peux pas utiliser la méthode `getCurrentSheet()` qui est dans le constructor.
`Call to an undefined method Box\Spout\Writer\WriterInterface::getCurrentSheet().`